code/modules/client/client procs.dm

```diff
@@ -362,14 +362,17 @@
 
 /client/proc/log_client_to_db()
 
-	if( IsGuestKey(src.key) )
+	if( IsGuestKey(key) )
 		return
 
+	// Isn't the ckey sanitized by default?
+	var/sql_ckey = sanitizeSQL(ckey)
```

Comment:
It's a ckey. It's already reduced to lowercase alphanumerics, underscores, maybe dashes; it's designed so you can use it, directly, as a filename for a player-specific save file. It's guaranteed, by BYOND, to contain nothing that would require sanitization.